lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -281,6 +281,48 @@ static LogicalResult drop(const AnnoPathValue &target, DictionaryAttr anno,
 // Customized Appliers
 //===----------------------------------------------------------------------===//
 
+static LogicalResult applyConventionAnno(const AnnoPathValue &target,
+                                         DictionaryAttr anno,
+                                         ApplyState &state) {
+  auto *op = target.ref.getOp();
+  auto loc = op->getLoc();
+  auto error = [&]() {
+    auto diag = mlir::emitError(loc);
+    diag << "firrtl.linkage.Convention ";
+    return diag;
+  };
+
+  auto opTarget = target.ref.dyn_cast<OpAnnoTarget>();
+  if (!opTarget)
+    return error() << "must target a module object";
+
+  if (!target.isLocal())
+    return error() << "must be local";
+
+  auto conventionStrAttr = anno.getAs<StringAttr>("convention");
+  if (!conventionStrAttr)
+    return error() << "missing convention field";
+
+  auto conventionStr = conventionStrAttr.getValue();
+  auto conventionOpt = symbolizeConvention(conventionStr);
+  if (!conventionOpt)
+    return error() << "unknown convention" << conventionStr;
```

Comment:
Not critical, but this reminds me that we should eventually come up with a way of auto-generating the unpacking of this as opposed to always doing this checking. 

You can consider using `tryGetAs` (https://github.com/llvm/circt/blob/fef7ca7d30e3a2a49ddfe6ad2fbd48da7cda5289/include/circt/Dialect/FIRRTL/FIRRTLAnnotationHelper.h#L304) to standardize some of the checking.